tests/function/test_config.py

```diff
@@ -301,3 +306,172 @@ def test_log_dir_and_log_file_in_virtwho_conf(self, virtwho, globalconf, hypervi
+            in /etc/virt-who.conf
+            2. Run virt-who with the correct owner setting in [defaults] section
+            in /etc/virt-who.conf
+
+        :expectedresults:
+
+            1. Virt-who runs failed with the incorrect owner setting
+            2. Succeeded to run the virt-who with correct owner setting
+        """
+        guest_uuid = hypervisor_data['guest_uuid']
+        globalconf.update('global', 'debug', 'True')
+        hypervisor.delete('owner')
+
+        globalconf.update('defaults', 'owner', owner_data['bad_owner'])
+        result = virtwho.run_service()
+        assert (result['error'] is not 0
+                and result['send'] == 0
+                and result['thread'] == 1
+                and any(error in result['error_msg'] for error in owner_data['error']))
+
+        globalconf.update('defaults', 'owner', owner_data['owner'])
+        result = virtwho.run_service()
+        assert (result['error'] == 0
+                and result['send'] == 1
+                and result['thread'] == 1
+                and guest_uuid in result['log'])
+
+    def test_hypervisor_id_in_virtwho_conf(self, virtwho, globalconf, hypervisor, hypervisor_data,
+                                           register_data, rhsm):
+        """Test the hypervisor_id option in /etc/virtwho.conf
+
+        :title: virt-who: config: test hypervisor_id option
+        :id: fed463a6-9538-4242-9990-2e4995d1f473
+        :caseimportance: High
+        :tags: tier1
+        :customerscenario: false
+        :upstream: no
+        :steps:
+
+            1. Run virt-who with hypervisor_id=hostname in /etc/virt-who.conf
+            2. Run virt-who with hypervisor_id=uuid in /etc/virt-who.conf
+            3. Run virt-who with hypervisor_id=hwuuid in /etc/virt-who.conf
+
+        :expectedresults:
+
+            1. Succeeded to run the virt-who, the hypervisor_id in mapping info should be hostname
+            2. Succeeded to run the virt-who, the hypervisor_id in mapping info should be uuid
+            3. Succeeded to run the virt-who, the hypervisor_id in mapping info should be hwuuid
+        """
+        globalconf.update('global', 'debug', 'True')
+        # we default have the hypervisor_id section in the config file in /etc/virt-who.d/
+        hypervisor.delete('hypervisor_id')
+
+        hypervisor_ids = ['hostname', 'uuid']
+        if HYPERVISOR in ['esx', 'rhevm']:
+            hypervisor_ids.append('hwuuid')
+        for hypervisor_id in hypervisor_ids:
+            # only esx and rhevm modes support hwuuid
+            globalconf.update('defaults', 'hypervisor_id', hypervisor_id)
+            result = virtwho.run_service()
+            assert (result['error'] == 0
+                    and result['send'] == 1
+                    and result['thread'] == 1
+                    and result['hypervisor_id'] == hypervisor_data[f'hypervisor_{hypervisor_id}'])
+
+        if REGISTER is 'rhsm':
```

Comment:
- If this `if` branch should be in the `for` loop?
- And as discussion we should define new functions to make sure the hypervisor entry can be shown successfully in the register server web and for satellite there should only one entry existing.